cumulus/parachain-template/node/src/service.rs

```diff
@@ -40,7 +40,10 @@ use substrate_prometheus_endpoint::Registry;
 pub struct ParachainNativeExecutor;
 
 impl sc_executor::NativeExecutionDispatch for ParachainNativeExecutor {
-	type ExtendHostFunctions = frame_benchmarking::benchmarking::HostFunctions;
+	type ExtendHostFunctions = (
+		cumulus_client_service::storage_proof_size::HostFunctions,
```

Comment:
If we propose this host function in the template. Maybe we should also need to add the `StorageWeigthReclaim` signed the runtime signed extensions tuple? (i.e. [here](https://github.com/paritytech/polkadot-sdk/blob/966a886439f4c5ef0bc81f32b8a189d758a2505f/cumulus/parachain-template/runtime/src/lib.rs#L101))

Otherwise what is its usage?

Reply:
> IIUC the StorageWeightReclaimer alone can't work correctly if we don't also return the extra weight to frame_system::BlockWeight ( by means of this signed extension hooks right?
Or I'm missing some other means of doing this? E.g. the user directly altering frame_system::BlockWeight in some other hook.

The `StorageWeightReclaimer` is a separate mechanism from the `SignedExtension`. It is designed to be used inside the `on_idle` hook, where weight management happens manually via `WeightMeter`. Users need to check whether the operation they plan fits into the remaining weight. In the end, `on_idle` will return the consumed weight and it will be accounted for after the hook finished here:
https://github.com/paritytech/polkadot-sdk/blob/150a360c410d9a051ac04022bb69eb1458396db6/substrate/frame/executive/src/lib.rs#L607-L614

This means that the weightreclaimer should only modify the value in the weightmeter, not the storage value itself. 

> If we propose this host function in the template. Maybe we should also need to add the StorageWeigthReclaim signed extension to the runtime signed extensions tuple? (i.e. [here](https://github.com/paritytech/polkadot-sdk/blob/966a886439f4c5ef0bc81f32b8a189d758a2505f/cumulus/parachain-template/runtime/src/lib.rs#L101))
>
> Otherwise what is its usage?

This is a good point, I will add it.